Utilities/StorageFactory/src/StorageAccount.cc

```diff
@@ -50,23 +56,50 @@ static std::string d2str(double d) {
   return t.str();
 }
 
+
+void StorageAccount::aggregateStatistics() {
+  std::set<std::string> protocols;
+  for (auto const &item : s_nameToToken) {
+    protocols.insert(item.first.second);
+  }
+  for (auto const &protocol : protocols) {
+    StorageAccount::setLabel("");
+    auto token = tokenForStorageClassName(protocol);
```

Comment:
I think `tokenForStorageClassName` should be changed to take both the protocol and the label. Then a new function should be added `tokenForStorageClassNameUsingContext` which takes only the protocol and gets the label from the thread local. So all existing calls to `tokenForStorageClassName` would need to be changed to the new name.